src/error.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+//
+// This program is licensed to you under the Apache License Version 2.0,
+// and you may not use this file except in compliance with the Apache License Version 2.0.
+// You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the Apache License Version 2.0 is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+use std::fmt::{Display, Formatter, Result};
+
+/// The errors that can occur when using the Snowplow Tracker
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum SnowplowError {
```

Comment:
Tiny tiny nit: Just looking at the reqwest library, I see that their error struct is just called `Error` and since it is hidden inside the `reqwest` crate, it doesn't cause conflicts. Maybe we could have a `snowplow_tracker::Error` as well?

Reply:
Yeah, I went back and forth deciding if it should be `SnowplowError` or `Error`, as `Error` seemed less "useful". Having another look around though it does seem that `Error` is the more standard naming, so the user can either use the namespace or rename the import with `use snowplow_tracker::Error as SnowplowError;`